Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,89 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+def is_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+def main():
+    internal_domains_list_name = demisto.args().get("InternalDomainsListName", "InternalDomains")
+    domains_to_check = argToList(demisto.args().get("Domains", ()))
+
+    # Get the list of internal domains from the XSOAR list:
+    internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list_name})[0]['Contents']
+    if "Item not found" in internal_domains:
+        return_error(f"The list name {internal_domains_list_name} does not exist.")
```

Comment:
Please move after the check if 'internal_domains' exists or not

Reply:
Please note there are 2 separate scenarios here. FIrst one - "Item not found" means the list does not exist in XSOAR. Then I return error.
The second scenario is: the list exists in XSOAR, but it's empty. Then I `return_results` that "no internal domains were found [for that list]...".
Hope this clears up the confusion